web/prisma/migrations/20240307175726_score_add_source/migration.sql

```diff
@@ -0,0 +1,15 @@
+-- CreateEnum
+CREATE TYPE "ScoreSource" AS ENUM ('API', 'REVIEW');
+
+-- Add a nullable column first
+ALTER TABLE "scores" ADD COLUMN "source" "ScoreSource";
+
+-- Set default values conditionally with explicit casting
+UPDATE "scores"
+SET "source" = CASE
+    WHEN "name" = 'manual-score' THEN 'REVIEW'::"ScoreSource"
+    ELSE 'API'::"ScoreSource"
+END;
+
+-- Alter column to not null after setting default values
+ALTER TABLE "scores" ALTER COLUMN "source" SET NOT NULL;
```

Comment:
This migration:
- locks the entire table during the migration. Ideally, we iterate over the table in chunks and update.
- removes the set not null constraint after updating. Between the two steps mentioned above, we could have added a new score in the meantime. In this case, the migration not null constraint would fail.

proposal:
- add another migration, in which:
  -  we get a full lock of the table
  - update the entire column again where the type is null
  - add the non null constraint
  - release the lock.

Reply:
replaced by two steps:

step 1: https://github.com/langfuse/langfuse/pull/1365/files
step 2: https://github.com/langfuse/langfuse/pull/1369